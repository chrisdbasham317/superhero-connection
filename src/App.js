import React, { Component } from 'react';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      isLoading: true,
      img: ''
    }
  }

  // componentDidMount = async () => {
  //   await fetch('https://www.superherodb.com/pictures2/portraits/10/100/791.jpg')
  //     .then(res => res.json())
  //     .then(data => {
  //       return this.setState({
  //         isLoading: false,
  //         img: data
  //       })
  //     })
  // }

  render() {
    return (
      <img src='https://www.superherodb.com/pictures2/portraits/10/100/791.jpg' />
    )
}
}

export default App;
