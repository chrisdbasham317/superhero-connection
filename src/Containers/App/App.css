body {
  font-family: 'McLaren', cursive;
}

.header--nav-bar {
  align-items: flex-end;
  background-image: url('https://static.skillshare.com/uploads/discussion/tmp/c18a7b07');
  display: flex;
  color: #000000;
  height: 15vh;
  justify-content: space-between;
  -webkit-text-stroke: 1px #ffffff;
}

.h1--main-title {
  font-size: 3rem;
  margin: 0px;
  padding-left: 2vw;
}

.div--nav-links {
  display: flex;
  font-size: 1.5rem;
  justify-content: space-around;
  width: 25%;
}

.links {
  text-decoration: none;
}