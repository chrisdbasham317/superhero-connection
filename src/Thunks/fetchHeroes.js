import { isLoading, hasErrored, setHeroes } from '../Actions';


export const fetchHeroes = () => {
  const defaultHeroIds = [38, 60, 63, 69, 76, 93, 95, 98, 204, 230, 459, 413, 387, 263, 298, 306, 579, 601, 632, 644, 642]
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const heroFetch = defaultHeroIds.map(id => {
        const response = await fetch(`https://superheroapi.com/api/10157857262907049/${id}`)
        if (!response.ok) {
          throw Error(response.statusText)
        }
      });
      const heroes = await heroFetch.json();
      dispatch(isLoading(false));
      dispatch(setHeroes(heroes));
    } catch {
      dispatch(isLoading(false));
      dispatch(hasErrored(error.message))
    }
  }
}

