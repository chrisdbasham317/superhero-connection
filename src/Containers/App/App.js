import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route, NavLink } from 'react-router-dom';
import { fetchHeroes } from '../../Thunks/fetchHeroes';
import Heroes from '../Heroes/Heroes';
import BattleGround from '../BattleGround/BattleGround';
import HeroPage from '../HeroPage/HeroPage';
import WinnerCircle from '../WinnerCircle/WinnerCircle';
import './App.css';

export class App extends Component {
  constructor() {
    super();

  }

  componentDidMount() {
    this.props.fetchHeroes();
  }

  render() {
    return (
      <div className='div--app'>
        <header className='header--nav-bar'>
          <h1>SuperHero Battle</h1>
          <div className='div--nav-links'>
            <NavLink className='links' to='/'>Home</NavLink>
            <NavLink className='links' to='/battle'>Battle</NavLink>
          </div>
        </header>
        <main>
          <Route exact path='/'>
            <Heroes />
          </Route>
          <Route path='/battle'>
            <BattleGround /> 
          </Route>
          <Route path='/hero/:id'>
            <HeroPage />
          </Route>
          <Route path='/winner'>
            <WinnerCircle />
          </Route>
        </main>
      </div>
    )
}
}

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({ fetchHeroes }, dispatch)
)

export default connect(null, mapDispatchToProps)(App);
