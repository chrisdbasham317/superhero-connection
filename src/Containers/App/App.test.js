import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { fetchHeroes } from '../../Thunks/fetchHeroes';


jest.mock('../../Thunks/fetchHeroes');

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <App
        fetchHeroes={jest.fn}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleClick is called', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleClick(mockEvent);
    expect(wrapper.state().modalState).toEqual(false);
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with an addTodo action when handleSubmit is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchHeroes();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchHeroes(actionToDispatch);

      // Expectaion
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
})