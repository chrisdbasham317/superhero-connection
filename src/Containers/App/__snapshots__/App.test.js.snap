// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="div--app"
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="WelcomeModal"
    closeTimeoutMS={0}
    contentLabel="App Welcome Page"
    isOpen={true}
    overlayClassName="WelcomeOverlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {},
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.85)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
        },
      }
    }
  >
    <article
      className="article--welcome-modal"
    >
      <h1
        className="h1--welcome-modal"
      >
        Welcome to SuperHero Battle!
      </h1>
      <p
        className="p--welocome-modal"
      >
        <span>
          SuperHero Battle
        </span>
         is a place where you can find out more information about some of your favorite superheroes and villians. More importantly, it's where you find out who could beat who in a head to head fight!
      </p>
      <button
        className="button--close-modal"
        onClick={[Function]}
      >
        Enter to find out!
      </button>
    </article>
  </Modal>
  <header
    className="header--nav-bar"
  >
    <h1
      className="h1--main-title"
    >
      SuperHero Battle
    </h1>
    <div
      className="div--nav-links"
    >
      <NavLink
        activeClassName="active"
        className="links"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
      <NavLink
        activeClassName="active"
        className="links"
        to="/battle"
      >
        Battle
      </NavLink>
    </div>
  </header>
  <main>
    <Route
      exact={true}
      path="/"
    >
      <Connect(Heroes) />
    </Route>
    <Route
      path="/battle"
    >
      <Connect(BattleGround) />
    </Route>
    <Route
      path="/hero/:id"
      render={[Function]}
    />
    <Route
      path="/winner"
    >
      <Connect(WinnerCircle) />
    </Route>
  </main>
</div>
`;
